Wrappers/Python/cil/optimisation/algorithms/ADMM.py

```diff
@@ -111,21 +111,18 @@ def update(self):
         
         self.tmp_adj *= -(self.tau/self.sigma)
         self.x += self.tmp_adj     
+      
         # apply proximal of f        
-        tmp = self.f.proximal(self.x, self.tau)
-        self.x.fill(tmp)
-        del tmp
-        
-        self.operator.direct(self.x, out = self.tmp_dir)  
-        self.u += self.tmp_dir
+        self.f.proximal(self.x, self.tau, out=self.x)
```

Comment:
Is this not reintroducing the bug?

self.x should not be the out container and one of the inputs.

Reply:
absolutely this needs to be reverted